[vulkan] Add core graph components

SS-JIA opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #92994

Differential Revision: [D42614155](https://our.internmc.facebook.com/intern/diff/D42614155/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D42614155/)!

Commit:
[vulkan] Add core graph components



Differential Revision: [D42614155](https://our.internmc.facebook.com/intern/diff/D42614155/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D42614155/)!

[ghstack-poisoned]

Commit:
Update on "[vulkan] Add core graph components"


Differential Revision: [D42614155](https://our.internmc.facebook.com/intern/diff/D42614155/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D42614155/)!

[ghstack-poisoned]

Skylion007 commented on aten/src/ATen/native/vulkan/graph/Value.h:
```diff
@@ -0,0 +1,141 @@
+    TensorStaging as_staging;
+
+    Payload() : u() {}
+    ~Payload() {}
```
```suggestion
    ~Payload() = default;
```

SS-JIA replied:
good point, adding that as well.

Skylion007 commented on aten/src/ATen/native/vulkan/api/Tensor.cpp:
```diff
@@ -266,6 +268,47 @@ vTensor::vTensor(
+      is_quantized_{other.is_quantized_},
+      q_scale_{other.q_scale_},
+      q_zero_point_{other.q_zero_point_},
+      view_(std::move(other.view_)) {}
```
Why not add move assignment for completeness?

SS-JIA replied:
ended up adding a defaulted move assignment as well

Skylion007 commented on aten/src/ATen/native/vulkan/api/Tensor.cpp:
```diff
@@ -2,6 +2,8 @@
 #include <ATen/native/vulkan/api/Utils.h>
 #include <c10/util/accumulate.h>
 
+#include <iostream>
```
Do you still need this include?

SS-JIA replied:
good catch, removed

Commit:
Update on "[vulkan] Add core graph components"


Differential Revision: [D42614155](https://our.internmc.facebook.com/intern/diff/D42614155/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D42614155/)!

[ghstack-poisoned]

Commit:
Update on "[vulkan] Add core graph components"


Differential Revision: [D42614155](https://our.internmc.facebook.com/intern/diff/D42614155/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D42614155/)!

[ghstack-poisoned]

Commit:
Update on "[vulkan] Add core graph components"


Differential Revision: [D42614155](https://our.internmc.facebook.com/intern/diff/D42614155/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D42614155/)!

[ghstack-poisoned]

Commit:
Update on "[vulkan] Add core graph components"


Differential Revision: [D42614155](https://our.internmc.facebook.com/intern/diff/D42614155/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D42614155/)!

[ghstack-poisoned]

Commit:
Update on "[vulkan] Add core graph components"


Differential Revision: [D42614155](https://our.internmc.facebook.com/intern/diff/D42614155/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D42614155/)!

[ghstack-poisoned]

Commit:
Update on "[vulkan] Add core graph components"


Differential Revision: [D42614155](https://our.internmc.facebook.com/intern/diff/D42614155/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D42614155/)!

[ghstack-poisoned]

salilsdesai reviewed (approved):
Lgtm

Commit:
Update on "[vulkan] Add core graph components"


Differential Revision: [D42614155](https://our.internmc.facebook.com/intern/diff/D42614155/)

**NOTE FOR REVIEWERS**: This PR has internal Meta-specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D42614155/)!

[ghstack-poisoned]

SS-JIA commented:
Unable to sync with Phabricator diff anymore via `ghexport`. Will close this PR and export it again into another PR.

SS-JIA commented:
New PR at: https://github.com/pytorch/pytorch/pull/94221
